core/trino-spi/src/main/java/io/trino/spi/Experimental.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.CONSTRUCTOR;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Designates an SPI that is still under development and is considered
+ * experimental. Experimental SPIs may be changed at any time and maybe removed.
+ */
+@Retention(RUNTIME)
+@Target({TYPE, FIELD, METHOD, CONSTRUCTOR})
+public @interface Experimental
+{
+    /**
+     * When the community believes the SPI will be finalized, but this date may be extended.
+     */
+    String until();
```

Comment:
I'm not sure this is a good idea. Who's going to be responsible to keep track of it? It will be quite confusing having a bunch of spis where this is in the past.

Reply:
I'd like to have a slack bot that annoys us when Experimental is expiring.

Reply:
How about creating a GitHub issue for this and adding a specific label to them?